system/src/system_sleep.cpp

```diff
@@ -106,8 +118,11 @@ void system_sleep(Spark_Sleep_TypeDef sleepMode, long seconds, uint32_t param, v
 
 void system_sleep_pin(uint16_t wakeUpPin, uint16_t edgeTriggerMode, long seconds, uint32_t param, void* reserved)
 {
-    network_suspend();
+	bool network_sleep = (param & 1)==0;
```

Comment:
Yep, I had planned to do this as part of the refactoring while writing the code - seems that fell through the cracks. Good catch!